lib/nexpose/scheduled_maintenance.rb

```diff
@@ -0,0 +1,83 @@
+module Nexpose
+# Constants useful across the Nexpose module.
+# Configuration structure for scheduled maintenance.
+  class ScheduledMaintenance < APIObject
+    require 'json'
+    include JsonSerializer
+
+    # Whether or not this maintenance schedule is enabled.
+    attr_accessor :enabled
+    # Valid schedule types: daily, hourly, monthly-date, monthly-day, weekly.
+    attr_accessor :schedule_type
+    # The repeat interval based upon type.
+    attr_accessor :schedule_interval
+    # The earliest date to generate the report on (in ISO 8601 format).
+    attr_accessor :schedule_start
+    # Whether the reindex task should run
+    attr_accessor :reindex
+    # Whether the compression task should run
+    attr_accessor :compress
+    # Whether the cleanup task should run
+    attr_accessor :cleanup
+    # Whether the maintenance should pause all local scans or wait for local scans to complete.
+    attr_accessor :pause_local_scans
+    # Cancellation window in minutes for scheduled DB maintenance.
+    attr_accessor :cancellation_period
+
+    def initialize(start, enabled=true, type, interval, reindex, compress, cleanup, pause_local_scans, cancellation_period)
```

Comment:
Same comment here, some suggested defaults:

* enabled=true
* reindex=true
* compress=true
* cleanup=true
* pause_local_scans=true
* cancellation_period=0 or nil (whichever is accepted for "not set")